container/sagemaker/proxy_client.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import grpc
+import logging
+
+from google.protobuf import text_format
+from tensorflow_serving.apis import model_service_pb2_grpc
+from tensorflow_serving.apis import model_management_pb2
+from tensorflow_serving.config import model_server_config_pb2
+
+# TFS_GRPC_PORT = os.environ['TFS_GRPC_PORT']
+MODEL_CONFIG_FILE = '/sagemaker/model-config.cfg'
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class GRPCProxyClient(object):
+    def __init__(self, port, host='0.0.0.0'):
+        self.channel = grpc.insecure_channel('{}:{}'.format(host, port))
+        self.stub = model_service_pb2_grpc.ModelServiceStub(self.channel)
+
+    def add_model(self, model_name, base_path, model_platform='tensorflow'):
```

Comment:
this function isn't safe for concurrent requests. is it possible for two add_model requests to arrive in close succession?

Reply:
That's a good point. I'll restrict access to the channel.